File: app/src/main/java/com/jsh/kr/alltest/ui/BaseFragment.java
package com.jsh.kr.alltest.ui;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.jsh.kr.alltest.util.LogUtil;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

public class BaseFragment extends Fragment {
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        LogUtil.d(this.getClass().getSimpleName(), "onCreate()");
    }

    @Override
    public void onResume() {
        super.onResume();
        LogUtil.d(this.getClass().getSimpleName(), "onResume()");
    }

    @Override
    public void onPause() {
        super.onPause();
        LogUtil.d(this.getClass().getSimpleName(), "onPause()");
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        LogUtil.d(this.getClass().getSimpleName(), "onCreateView()");
        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        LogUtil.d(this.getClass().getSimpleName(), "onAttach()");
    }

    @Override
    public void onDetach() {
        super.onDetach();
        LogUtil.d(this.getClass().getSimpleName(), "onDetach()");
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        LogUtil.d(this.getClass().getSimpleName(), "onDestroy()");
    }
}

File: appkt/src/main/java/com/jsh/kr/alltestkt/ui/BaseFragment.kt
package com.jsh.kr.alltestkt.ui

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.annotation.StringRes
import androidx.fragment.app.Fragment
import com.jsh.kr.alltestkt.util.LogUtil

open class BaseFragment : Fragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        LogUtil.d(this.javaClass.simpleName, "onCreate()")
    }

    override fun onStart() {
        super.onStart()
        LogUtil.d(this.javaClass.simpleName, "onStart()")
    }

    override fun onResume() {
        super.onResume()
        LogUtil.d(this.javaClass.simpleName, "onResume()")
    }

    override fun onPause() {
        super.onPause()
        LogUtil.d(this.javaClass.simpleName, "onPause()")
    }

    override fun onStop() {
        super.onStop()
        LogUtil.d(this.javaClass.simpleName, "onStop()")
    }

    override fun onDestroy() {
        super.onDestroy()
        LogUtil.d(this.javaClass.simpleName, "onDestroy()")
    }

    override fun onDestroyView() {
        super.onDestroyView()
        LogUtil.d(this.javaClass.simpleName, "onDestroyView()")
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        LogUtil.d(this.javaClass.simpleName, "onAttach()")
    }

    override fun onDetach() {
        super.onDetach()
        LogUtil.d(this.javaClass.simpleName, "onDetach()")
    }

}